File: dotnet/Controllers/Api/TestController.cs
namespace Fearless.Controllers.Api;

using Microsoft.AspNetCore.Mvc;
using static Utils.MyApiResponse;


[ApiController]
[Route("api/[controller]")]
public class TestController : ControllerBase
{
  [HttpGet]
  public  IResult Get(){
    
    var data = new { 
      user_ip_address = Ip ,
      current_url = CurrentUrl
    };
    
    return myApiResponse( data: data ,  code:200  );
  }

  public string Ip
  {
    get {
      var remoteIp = HttpContext.Connection.RemoteIpAddress;
      if (remoteIp != null && remoteIp.IsIPv4MappedToIPv6)
      {
        return remoteIp.MapToIPv4().ToString();
      }
      else 
        return remoteIp?.ToString() ?? "" ;
    }
  }

  public string CurrentUrl
  {
    get {
      var url = $"{HttpContext.Request.Host}{HttpContext.Request.Path}";
      return url;
    }
  }
}
 

File: nestjs/src/controllers/testcontroller/test.controller.ts
import { myApiResponse } from '@Fearless/utils/helpers';
import { Controller, Get, Req } from '@nestjs/common';
import { Request } from 'express';

@Controller('api/test')
export class TestController {

  @Get()
  index( @Req() req: Request ){

    const data = {
      user_ip_address: `${ req.ip }`,
      current_url: `${req.protocol}://${req.get('Host')}${req.originalUrl}`,
    }
    return myApiResponse({ data , code:200 })
  }
}
